### PR TITLE
addded-search-feature

### PR DESCRIPTION
**Pull request by CharlieMantium**

### CODE REVIEW

File: src/components/ActivitiesListItem/ActivitiesListItem.js [hunk 1]
```diff
@@ -27,29 +30,53 @@ const ActivityDesc = styled.p`
...
-        id="planView.activityDetails"
-        defaultMessage="Teacher: {teacher}, in room: {room}"
-        values={{ teacher, room }}
-        data-test="activity-details"
-      />
-    </ActivityDesc>
-  </>
-);
+const HighlightedSpan = styled.span`
+  color: ${props => props.isHighlighted && `${colors.pink}`};
```
Comment (Lukasz-Semik) [inline src/components/ActivitiesListItem/ActivitiesListItem.js]: I am not the fan of naming colors by they actual color. If you will change pink to green... You will have to change many places.

File: src/helpers/divideByFilteredString.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+export const divideByFilteredString = (inputText, searchedText) => {
```
Comment (Lukasz-Semik) [inline src/helpers/divideByFilteredString.js]: Did you try to find the solution for your case ? :) 
I think, that this function needs some tests :)

File: src/helpers/divideByFilteredString.js [hunk 3]
```diff
@@ -21,6 +21,8 @@ export const divideByFilteredString = (inputText, searchedText) => {
       tempString.toLowerCase().indexOf(searchedText.toLowerCase()) + searchedLength,
     );
   }
-  dividedTextArray.push({ str: tempString, isSearched: false });
+  if (tempString) {
```
Comment (Lukasz-Semik) [inline src/helpers/divideByFilteredString.js]: hm... so there was an edge case :D ?

File: src/helpers/divideByFilteredString.test.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { strings } from 'tests/fixtures/strings';
+
+import { divideByFilteredString } from './divideByFilteredString';
+
+describe('divideByFilteredString', () => {
+  it('should return the correct values', () => {
+    strings.forEach(testCase => {
+      const { inputString } = testCase;
```
Comment (Lukasz-Semik) [inline src/helpers/divideByFilteredString.test.js]: `const { inputString, searchedString, expectedResult } = testCase;`

File: src/tests/fixtures/strings.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
+export const strings = [
+  {
```
Comment (Lukasz-Semik) [inline src/tests/fixtures/strings.js]: I would simulate the searched string values:
`null`, `some number`, `some weird value like fucked up object` and `undefined`, but in the way, that you are invoking function without any argument.

Comment (Lukasz-Semik) [reply src/tests/fixtures/strings.js]: I mean, as an addition to thing you already assumed :)

File: src/helpers/divideByFilteredString.js [hunk 6]
```diff
@@ -1,5 +1,7 @@
-export const divideByFilteredString = (inputText, searchedText) => {
+export const divideByFilteredString = (inputText, inputSearched) => {
+  if (inputSearched === null || inputSearched === undefined || inputSearched === '') return [];
```
Comment (Lukasz-Semik) [inline src/helpers/divideByFilteredString.js]: `if (inpuSearched)` should be enough. 
to be super explicit two first conditions can be changed to `inputSearched == null` (coersion will do also validation to undefined.
